pkg/controllers/manifestwork.go

```diff
@@ -69,11 +70,34 @@ func ScaffoldManifestwork(hyd *hypdeployment.HypershiftDeployment) (*workv1.Mani
 					hyd.GetName()),
 			},
 		},
-		Spec: workv1.ManifestWorkSpec{},
-	}
-
-	if hyd.Spec.Override == hypdeployment.InfraOverrideDestroy {
-		w.Spec.DeleteOption = &workv1.DeleteOption{PropagationPolicy: workv1.DeletePropagationPolicyTypeOrphan}
+		Spec: workv1.ManifestWorkSpec{
+			Workload: workv1.ManifestsTemplate{
+				Manifests: []workv1.Manifest{
+					{
+						RawExtension: runtime.RawExtension{Object: &corev1.Namespace{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: targetNamespace,
+							},
+							TypeMeta: metav1.TypeMeta{
+								Kind:       "Namespace",
+								APIVersion: corev1.SchemeGroupVersion.String(),
+							},
+						}},
+					},
+				},
+			},
+			DeleteOption: &workv1.DeleteOption{
+				PropagationPolicy: workv1.DeletePropagationPolicyTypeSelectivelyOrphan,
+				SelectivelyOrphan: &workv1.SelectivelyOrphan{
+					OrphaningRules: []workv1.OrphaningRule{
```

Comment:
@jnpacker should we mark hostedcluster and nodepool as orphan as welll to prevent accident deletion by removing manifestwork?

Reply:
PTAL